spec/support/app_helpers.rb

```diff
@@ -1,6 +1,39 @@
 require "spec_helper"
 
 module AppHelpers
+  RSpec.shared_examples "json-only endpoint" do |base_path, query_parameters|
+    disallowed_formats = %w[.xml .html .txt .csv]
```

Comment:
I can [see we raise a 404](https://github.com/alphagov/search-api/blob/main/lib/rummager/app.rb#L82-L87) if the request is **not** JSON, but `disallowed_formats` sounds like a specific denylist. How have you come up with this list? Is it just reasonable examples that we might get, or is it defined somewhere?

Would you mind adding a comment if the former? And linking to the `json_only` method in the [commit](46f74a31e5518ba00184fe32b09929ed81673d90) message.